packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2147,10 +2293,15 @@ abstract class RenderBox extends RenderObject {
     super.markNeedsLayout();
   }
 
+  /// {@macro flutter.rendering.RenderObject.performResize}
+  ///
+  /// By default this method calls [getDryLayout] with the current
+  /// [constraints]. Instead of overriding this method, consider overriding
+  /// [computeDryLayout] (the backend implementation of [getDryLayout]).
   @override
   void performResize() {
     // default behavior for subclasses that have sizedByParent = true
-    size = constraints.smallest;
+    size = computeDryLayout(constraints);
```

Comment:
In the framework there appear to be a total of 16 RenderObject subclasses that set `sizedByParent` to true. Of those, only 2 used to rely on the default of `performResize` (constraints.smallest) and the remaining 14 had custom implementations. If I recall correctly, there also wasn't any customer_test that relied on the old default of `performResize`. With that, I wouldn't really call this "hugely breaking"...

I considered making the default computeDryLayout return `constraints.smallest` if `sizedByParent` was set, but ultimately decided against making that method more complicated (and harder to reason about) since the impact of the change seemed so low. 

Having said all that, is there something I missed that makes this change more breaking?

Reply:
Interesting.  When I ran across this, it's because I hit the error in my app; I had three render objects in which I had overridden `sizedByParent` to return true, and in all three cases, I was relying on the old default `performResize` behavior.

But based on your data, it seems I may be an outlier.

Reply:
Side-note: once my app's done (and I write tests), I plan to add them to the customer tests 🙂